include/boost/json/value.hpp

```diff
@@ -3266,6 +3271,121 @@ class value
+          an empty object and considered by the function again.
+        <br/>
+
+        The type of created intermediate elements is always @ref object. The
+        function creates at most one element for each pointer segment.
+
+        @par Complexity
+        Linear in the size of `ptr`.
+
+        @par Exception Safety
+        Basic guarantee.
+
+        @param ptr JSON Pointer string.
+
+        @param jv The value to assign to pointed element.
+
+        @return Pointer to the element identified by `ptr`.
+
+        @see
+        <a href="https://datatracker.ietf.org/doc/html/rfc6901">
+            RFC 6901 - JavaScript Object Notation (JSON) Pointer</a>
+    */
+    BOOST_JSON_DECL
+    value&
+    set_at_pointer(string_view ptr, value_ref jv);
+
+    /** Set an element via JSON Pointer.
+
+        This function is used to insert or assign to a potentially nested
+        element of the value using a JSON Pointer string. The function may
+        create intermediate elements corresponding to pointer segments.
+        <br/>
+
+        For a traversed pointer segment, if the corresponding element does
+        not exist or is a `null` value
+
+        @li if the containing value is an @ref object, a new
+          @ref key_value_pair is inserted with the segment's _unescaped_ string
+          as key and empty object as value; otherwise
+
+        @li if the containing value is an @ref array, then if the segment is
+          pass-the-end marker (`-`) or if it is a number equal to array's
+          size, then an empty object is appended; if the segment is larger
+          than the array's size, the function produces an error; otherwise
+
+        @li if the containing value is a `null` value, then it is replaced with
+          an empty object and considered by the function again.
+        <br/>
+
+        The type of created intermediate elements is always @ref object. The
+        function creates at most one element for each pointer segment.
+
+        @par Complexity
+        Linear in the size of `ptr`.
+
+        @par Exception Safety
+        Basic guarantee.
+        Calls to `memory_resource::allocate` may throw.
+
+        @param ptr JSON Pointer string.
+
+        @param jv The value to assign to pointed element.
+
+        @param ec Set to the error, if any occurred.
+
+        @return Pointer to the element identified by `ptr`.
+
+        @see
+        <a href="https://datatracker.ietf.org/doc/html/rfc6901">
+            RFC 6901 - JavaScript Object Notation (JSON) Pointer</a>
+    */
+/** @{ */
+    BOOST_JSON_DECL
+    value*
+    try_set_at_pointer(string_view ptr, value_ref jv, error_code& ec);
+
+    BOOST_JSON_DECL
+    value*
+    try_set_at_pointer(string_view ptr, value_ref jv, std::error_code& ec);
+/** @} */
+
+    //------------------------------------------------------
+
+    BOOST_JSON_DECL
+    const_pointer_traverse_state
+    step_over_pointer(string_view ptr) const noexcept;
```

Comment:
`ptr` is misleading, the convention in my libraries is to name the variable based on its type:

`jv` json::value
`s` string or string_view
`ec` error code

Only very rarely are things named after what they are, e.g. `string_view key`. But `ptr` is very suggestive of a pointer which it is not. This is misleading